app/assets/javascripts/admin/addon/components/watched-word-form.js

```diff
@@ -56,6 +56,11 @@ export default Component.extend({
     });
   },
 
+  @discourseComputed("word")
```

Comment:
maybe consider using this instead: https://api.emberjs.com/ember/release/functions/@ember%2Fobject%2Fcomputed/not